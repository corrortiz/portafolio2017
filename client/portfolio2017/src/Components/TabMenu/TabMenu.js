// @flow
import React, { Component } from 'react';

import { Link } from 'react-router-dom';

import Tabs, { Tab } from 'material-ui/Tabs';

import FavoriteIcon from 'material-ui-icons/Favorite';
import PersonPinIcon from 'material-ui-icons/PersonPin';
import HelpIcon from 'material-ui-icons/Help';
import ThumbUp from 'material-ui-icons/ThumbUp';

import GlobalsConnect from '../../HOC/GlobalsConnect/GlobalsConnect';

import { lenguajeSelector } from '../../Store/Actions/globals';
import { Home, Services, Projects, Contact } from '../../Assets/diccionary';

class TabMenu extends Component {
  state = {
    value: 0
  };

  handleChange = (event, value) => {
    this.setState(() => ({ value }));
  };

  render() {
    const { value } = this.state;
    const { lenguaje } = this.props.globals;

    return (
      <div className="header__bottonNavigation">
        <Tabs
          value={value}
          onChange={this.handleChange}
          scrollable
          scrollButtons="on"
          indicatorColor="primary"
          textColor="primary"
          className="header__bottonNavigation__tabs"
          centered
        >
          <Tab
            label={lenguajeSelector(lenguaje, Home)}
            icon={<FavoriteIcon />}
            component={Link}
            to={'/'}
          />
          <Tab
            label={lenguajeSelector(lenguaje, Services)}
            icon={<PersonPinIcon />}
            component={Link}
            to={'/services'}
          />
          <Tab
            label={lenguajeSelector(lenguaje, Projects)}
            icon={<HelpIcon />}
            component={Link}
            to={'/projects'}
          />
          <Tab
            label={lenguajeSelector(lenguaje, Contact)}
            icon={<ThumbUp />}
            component={Link}
            to={'/contact'}
          />
        </Tabs>
      </div>
    );
  }
}

export default GlobalsConnect(TabMenu);
