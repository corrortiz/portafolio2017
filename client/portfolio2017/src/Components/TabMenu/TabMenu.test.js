// @flow
import React from 'react';
import { shallow } from 'enzyme';
import TabMenu from './TabMenu';

it('should compare a snapshot of TabMenu', () => {
  const wrapper = shallow(<TabMenu />);
  expect(wrapper).toMatchSnapshot();
});
