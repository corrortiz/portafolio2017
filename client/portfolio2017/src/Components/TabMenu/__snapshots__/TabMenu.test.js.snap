// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compare a snapshot of TabMenu 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withStyles(Tabs)
        centered={true}
        indicatorColor="primary"
        onChange={[Function]}
        scrollButtons="on"
        scrollable={true}
        textColor="primary"
        value={0}
>
        <withStyles(Tab)
                component={[Function]}
                icon={<pure(Favorite) />}
                label="Home"
                to="/"
        />
        <withStyles(Tab)
                component={[Function]}
                icon={<pure(PersonPin) />}
                label="Services"
                to="/services"
        />
        <withStyles(Tab)
                component={[Function]}
                icon={<pure(Help) />}
                label="Projects"
                to="/projects"
        />
        <withStyles(Tab)
                component={[Function]}
                icon={<pure(ThumbUp) />}
                label="Contact"
                to="/contact"
        />
</withStyles(Tabs)>,
      "className": "header__bottonNavigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <withStyles(Tab)
            component={[Function]}
            icon={<pure(Favorite) />}
            label="Home"
            to="/"
/>,
          <withStyles(Tab)
            component={[Function]}
            icon={<pure(PersonPin) />}
            label="Services"
            to="/services"
/>,
          <withStyles(Tab)
            component={[Function]}
            icon={<pure(Help) />}
            label="Projects"
            to="/projects"
/>,
          <withStyles(Tab)
            component={[Function]}
            icon={<pure(ThumbUp) />}
            label="Contact"
            to="/contact"
/>,
        ],
        "indicatorColor": "primary",
        "onChange": [Function],
        "scrollButtons": "on",
        "scrollable": true,
        "textColor": "primary",
        "value": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "icon": <pure(Favorite) />,
            "label": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "icon": <pure(PersonPin) />,
            "label": "Services",
            "to": "/services",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "icon": <pure(Help) />,
            "label": "Projects",
            "to": "/projects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "icon": <pure(ThumbUp) />,
            "label": "Contact",
            "to": "/contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withStyles(Tabs)
          centered={true}
          indicatorColor="primary"
          onChange={[Function]}
          scrollButtons="on"
          scrollable={true}
          textColor="primary"
          value={0}
>
          <withStyles(Tab)
                    component={[Function]}
                    icon={<pure(Favorite) />}
                    label="Home"
                    to="/"
          />
          <withStyles(Tab)
                    component={[Function]}
                    icon={<pure(PersonPin) />}
                    label="Services"
                    to="/services"
          />
          <withStyles(Tab)
                    component={[Function]}
                    icon={<pure(Help) />}
                    label="Projects"
                    to="/projects"
          />
          <withStyles(Tab)
                    component={[Function]}
                    icon={<pure(ThumbUp) />}
                    label="Contact"
                    to="/contact"
          />
</withStyles(Tabs)>,
        "className": "header__bottonNavigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "centered": true,
          "children": Array [
            <withStyles(Tab)
              component={[Function]}
              icon={<pure(Favorite) />}
              label="Home"
              to="/"
/>,
            <withStyles(Tab)
              component={[Function]}
              icon={<pure(PersonPin) />}
              label="Services"
              to="/services"
/>,
            <withStyles(Tab)
              component={[Function]}
              icon={<pure(Help) />}
              label="Projects"
              to="/projects"
/>,
            <withStyles(Tab)
              component={[Function]}
              icon={<pure(ThumbUp) />}
              label="Contact"
              to="/contact"
/>,
          ],
          "indicatorColor": "primary",
          "onChange": [Function],
          "scrollButtons": "on",
          "scrollable": true,
          "textColor": "primary",
          "value": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "icon": <pure(Favorite) />,
              "label": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "icon": <pure(PersonPin) />,
              "label": "Services",
              "to": "/services",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "icon": <pure(Help) />,
              "label": "Projects",
              "to": "/projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "icon": <pure(ThumbUp) />,
              "label": "Contact",
              "to": "/contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
